pincer/objects/guild/channel.py

```diff
@@ -779,6 +779,69 @@ async def fetch_message(self, message_id: int) -> UserMessage:
             await self._http.get(f"channels/{self.id}/messages/{message_id}")
         )
 
+    async def history(
+        self, limit: int = 50,
+        before: Union[int, Snowflake] = None,
+        after: Union[int, Snowflake] = None,
+        around: Union[int, Snowflake] = None,
+    ) -> AsyncIterator[UserMessage]:
+        """
+        Returns a list of messages in this channel.
+
+        Parameters
+        ----------
+        around
+            The message ID to look around.
+        after
+            The message ID to look after.
+        before
+            The message ID to look before.
+        limit
+            The maximum number of messages to return.
+
+        Returns
+        -------
+        AsyncIterator[:class:`~pincer.objects.message.user_message.UserMessage`]
+            An iterator of messages.
+        """
+
+        if isinstance(before, int):
+            before = Snowflake(before)
+
+        if isinstance(after, int):
+            after = Snowflake(after)
+
+        if isinstance(around, int):
+            around = Snowflake(around)
+
+        if limit is None:
+            limit = float('inf')
+
+        while limit > 0:
+            retrieve = min(limit, 100)
```

Comment:
limit HAS to be capped to 100. If they choose a number too big there will be rate limits and that will cause  it to stall because of an "invisible error"

Reply:
I don't agree, we need to provide an user-friendly way to get all messages within a channel, even if it is extremely slow. 
Discord.py do it has well, see `channel.history(limit=...).flatten()`

Also it deal very nicely with ratelimiting.
I test if with 4000 messages, in `41.18647676300316 seconds`

Reply:
I see your point but you need to change things
1. Limit should never be set to infinity if the user doesn't explicitly do so. I would make it 100 to prevent rate limits from a mistake.
2. There can be a lot of api calls so you should use gather. You can even use itertools.chain around gather. I bet that would be useful here.